src/macros.php

```diff
@@ -297,3 +299,49 @@
         return $this->reverse()->after($currentItem, $fallback);
     });
 }
+
+if (! Collection::hasMacro('paginate')) {
+    /*
+     * Paginate the given collection
+     *
+     * @param int $perPage
+     * @param int $total
+     * @param int $page
+     * @param string $pageName
+     *
+     * @return \Illuminate\Pagination\LengthAwarePaginator
+     */
+    Collection::macro('paginate', function (int $perPage = 15, string $pageName = 'page', int $page = null, int $total = null, array $options = []): LengthAwarePaginator {
+        $page = $page ?: LengthAwarePaginator::resolveCurrentPage($pageName);
+        $results = $this->forPage($page, $perPage);
+        $total = $total ?: $this->count();
+        $options += [
+            'path' => LengthAwarePaginator::resolveCurrentPath(),
+            'pageName' => $pageName,
+        ];
+
+        return new LengthAwarePaginator($results, $total, $perPage, $page, $options);
+    });
+}
+
+if (! Collection::hasMacro('simplePaginate')) {
+    /*
+     * Paginate the collection into a simple paginator
+     *
+     * @param int $perPage
+     * @param int $page
+     * @param string $pageName
+     *
+     * @return \Illuminate\Contracts\Pagination\Paginator
+     */
+    Collection::macro('simplePaginate', function (int $perPage = 15, string $pageName = 'page', int $page = null, array $options = []): Paginator {
```

Comment:
Please exactly match the `Paginator's signature for the method.

`$items, $perPage, $currentPage = null, array $options = []`